### PR TITLE
Slt 526/featured training cards

### PR DESCRIPTION
**Pull request by mattgraygithub**
Work in progress for adding the training cards to the training overview page for [Leankit Ticket 526](https://codurance-online.leankit.com/card/1151267429)

### CODE REVIEW

File: src/_includes/training_card.liquid [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{% if include.training.featured %}
+    {% capture avatar %}<img class="training-card__avatar" alt=""src={{ site.data.service_lines.training.avatar-src | prepend: site.baseurl }}>{% endcapture %}
+    {% assign author = include.training.author %}
+    {% capture tag %}<h2 class="training-card__tag">{{ avatar }} {{ author }} </h2>{% endcapture %}
```
Comment (LL782) [inline src/_includes/training_card.liquid:4]:
```suggestion
    {% capture tag %}<h2 class="training-card__tag">{{ avatar }} {{ author }}</h2>{% endcapture %}
```

File: src/_includes/training_card.liquid [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    {% assign featured_class = "training-card--featured" %}
+{% else %}
+    {% capture tag %}{% endcapture %}
+    {% assign featured_class = "" %}
```
Comment (LL782) [inline src/_includes/training_card.liquid:8]: I don't think you need either of these

```suggestion
{% else %}
    {% capture tag %}{% endcapture %}
    {% assign featured_class = "" %}
```

File: src/_includes/training_card.liquid [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+<article class="training-card {{ featured_class }}">
+    <div class="training-card__primary-content">
+        {{ tag }}
+        <h3 class="training-card__title">{{ include.training.name }}</h3>
+    </div>
+    <div class="training-card__secondary-content">
+        <p class="training-card__description">{{ include.training.description }}</p>
+        {% capture cta-href %}{{ include.training.page | prepend: site.baseurl }}{% endcapture %}
+        {% capture cta-text %}{% t services.learn-more %}{% endcapture %}
```
Comment (LL782) [inline src/_includes/training_card.liquid:19]: [Minor idea] Might be nice to put these two lines before the artcle tag, just to get all the heavy Liquid stuff out of the markup when possible.

File: src/_includes/training_card_group.liquid [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+<div class="training-card-group--only-featured">
+  {%- for training in training_data -%}
+    {%- if training.featured -%}
+      {% include training_card.liquid training=training %}
+    {%- endif -%}
+  {%- endfor -%}
+</div>
+
+<div class="training-card-group">
+  <div class="training-card-group__track" data-nav-track data-position="1" data-number-of-cards="4">
```
Comment (LL782) [inline src/_includes/training_card_group.liquid:16]: Is `data-number-of-cards` meant to be hard coded?

File: src/_includes/training_card_group.liquid [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    {% for training in training_data %}
+      {% include training_card.liquid training=training %}
+    {% endfor %}
+  </div>
+  <div class="training-card-group__navigation-container">
+    {%- assign non-featured-training = training_data | where_exp:"item", "item.featured != true" -%}
+    {%- assign counter = 1 -%}
+    {%- for training in non-featured-training -%}
+      {%- if forloop.first == true -%}{%- assign active-class = "active" -%}{%- else -%}{%- assign active-class = "" -%}{%- endif -%}
+      <div class="training-card-group__navigation-item {{active-class}}" data-nav-item data-position="{{counter}}"></div>
```
Comment (LL782) [inline src/_includes/training_card_group.liquid:26]:
```suggestion
        <div class="training-card-group__navigation-item {{active-class}}" data-nav-item data-position="{{counter}}"></div>
```

File: src/assets/custom/css/_sass/_service-line-hero.scss [hunk 6]
```diff
@@ -64,14 +64,15 @@ $medium-width-content-padding: 100px;
     padding-left: $medium-width-content-padding;
     padding-right: $medium-width-content-padding;
   }
+  @include large{
+    margin-top: -100px;
```
Comment (LL782) [inline src/assets/custom/css/_sass/_service-line-hero.scss:68]: How come this is being added? Make sure you get a design review especially of the service line pages. Also, there's a positive padding 50px which is then being overridden with negatvie margin. Starting to get messy

File: src/assets/custom/css/_sass/_training-card-group.scss [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+$space-around-cards-on-small-screen: 15px;
+$number-of-cards-in-carousel: 4;
+
+$total-width-of-cards-on-small-screen:
+  $width-of-cards-on-small-screen * $number-of-cards-in-carousel;
+
+$total-width-of-space:
+    $space-around-cards-on-small-screen * $number-of-cards-in-carousel;
+
+$complete-total-width:
```
Comment (LL782) [inline src/assets/custom/css/_sass/_training-card-group.scss:13]: Not sure about this name... maybe, `$width-of-track` might be better?

File: src/assets/custom/css/_sass/_training-card-group.scss [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+  $total-width-of-cards-on-small-screen + $total-width-of-space;
+
+.training-card-group {
+  @include small {
+    overflow: hidden;
+    padding-top: 45px;
+    width: 100%;
```
Comment (LL782) [inline src/assets/custom/css/_sass/_training-card-group.scss:20]: Is this width 100% needed?

File: src/assets/custom/css/_sass/_training-card-group.scss [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+  }
+}
+
+.training-card-group--only-featured {
+  @include small {
+    display: block;
+  }
```
Comment (LL782) [inline src/assets/custom/css/_sass/_training-card-group.scss:27]: Why do we need to set display block on a div?

```suggestion
```

File: src/assets/custom/css/_sass/_training-card-group.scss [hunk 10]
```diff
@@ -0,0 +1,133 @@
+$width-of-cards-on-small-screen: 310px;
+$half-of-card-width: ($width-of-cards-on-small-screen / 2);
+$space-around-cards-from-medium-width: 25px;
+$space-around-cards-on-small-screen: 15px;
+$number-of-cards-in-carousel: 4;
```
Comment (LL782) [inline src/assets/custom/css/_sass/_training-card-group.scss:5]: Ok, so for now this is still hardcoded... Cool. Simplest thing first

File: src/assets/custom/css/_sass/_training-card-group.scss [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+  }
+  @include medium-large-and-extra-large{
+    justify-content: space-between;
+  }
+}
+
+.training-card-group__track .training-card {
+  @include small {
+    width: $width-of-cards-on-small-screen;
+    box-shadow: 0px 0px 8px rgba(0,0,0,0.2);
```
Comment (LL782) [inline src/assets/custom/css/_sass/_training-card-group.scss:69]: Is this a different shadow to the one we already have?

File: src/assets/custom/css/_sass/_training-card-group.scss [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+      display: none;
+    }
+  }
+}
+
+.training-card-group__navigation-container {
+  padding-bottom: 45px;
+  padding-top: 45px;
+  text-align: center;
+  @include medium-large-and-extra-large {
```
Comment (LL782) [inline src/assets/custom/css/_sass/_training-card-group.scss:113]:
```suggestion
  text-align: center;
  
  @include medium-large-and-extra-large {
```

File: src/assets/custom/css/_sass/_training-card.scss [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+  @include small {
+    @include mercer-bold;
+  }
+  @include medium-large-and-extra-large {
+    @include freed-bold;
+  }
+
+  .training-card--featured & {
+    @include freed-bold;
+  }
```
Comment (LL782) [inline src/assets/custom/css/_sass/_training-card.scss:60]: Personally I find this slightly faster to read and it also avoids overrides in the CSS

Not a big deal, just an idea

```suggestion
  .training-card:not(.training-card--featured) & {
    @include small {
      @include mercer-bold;
    }
    @include medium-large-and-extra-large {
      @include freed-bold;
    }
  }

  .training-card--featured & {
    @include freed-bold;
  }
```

File: src/assets/custom/css/_sass/_training-card.scss [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+
+  @include small {
+    @include base;
+  }
+  @include medium-large-and-extra-large {
+    @include mercer-reg;
+  }
+
+  .training-card--featured & {
+    @include mercer-reg;
```
Comment (LL782) [inline src/assets/custom/css/_sass/_training-card.scss:74]: Same idea as mentioned above _could_ be applied here

File: src/assets/custom/css/_sass/_training-overview.scss [hunk 15]
```diff
@@ -3,15 +3,28 @@ $katalyst-banner-height--small-medium: 313px;
...
   @include max-width--out-there;
+  @include medium-large-and-extra-large {
+    @include lateral-spacing;
+  }
+}
+
+.training-overview__flexslider-wrapper {
+  @include medium-large-and-extra-large {
+    display: none;
+  }
```
Comment (LL782) [inline src/assets/custom/css/_sass/_training-overview.scss:27]: Is this being used?
```suggestion
```

**Review by LL782 — APPROVED** [review]
:+1: awesome work. All reads really nicely. Let's get this in

File: src/assets/custom/css/_sass/_training-overview.scss (hunk 15 above)
Comment (Dan-Bird) [reply src/assets/custom/css/_sass/_training-overview.scss:27]: This one is fixed in here: [e996f96](https://github.com/codurance/site/pull/1917/commits/e996f96320ae9c7682d5198dfec401fcc9471b2b)

**Pull request merged**